index.js

```diff
@@ -146,11 +146,12 @@ class ReaddirpStream extends Readable {
 
     this.filesToRead += files.length;
 
-    const entries = await Promise.all(files.map(dirent => this._formatEntry(dirent, parent)));
+    const promises = files.map(dirent => this._formatEntry(dirent, parent));
 
-    if (this.destroyed) return;
-
-    for (const entry of entries) {
+    for (const promise of promises) {
+      const entry = await promise
```

Comment:
You are still using `await` here. Shouldn't this be refactored further?

Reply:
> If order doesn't matter this could be further improved.

If order matters... then I don't think so...

Reply:
@paulmillr does the order matter here?

@ronag there are 2 more cases in test.js, can you have a look at that too? I'm not sure if the order matter there either, though

Reply:
Yeah kinda. I want to test the perf though

Reply:
I don't think there is a big difference. The first entry should in theory be the one to complete first unless there is a cache or something involved on the os side.